test.disabled/ejabberd_tests/tests/push_integration_SUITE.erl

```diff
@@ -0,0 +1,553 @@
+            escalus:send(Bob, Stanza),
+
+            [Notification = #{}] = get_push_logs(fcm, DeviceToken, Config),
+
+            FCMData = maps:get(<<"notification">>, maps:get(<<"request_data">>, Notification)),
+            ?assertMatch(#{<<"body">> := <<"Heyah!">>}, FCMData),
+            ?assertMatch(#{<<"title">> := SenderJID}, FCMData),
+            ?assertMatch(#{<<"tag">> := SenderJID}, FCMData),
+            ?assertMatch(#{<<"click_action">> :=  <<"myactivity">>}, FCMData),
+
+            ok
+        end).
+
+
+%%--------------------------------------------------------------------
+%% Test helpers
+%%--------------------------------------------------------------------
+
+get_push_logs(Service, DeviceToken, Config) ->
+    PushMock = connect_to(Service),
+    wait_for(timer:seconds(10), fun() ->
+        {ok, 200, Body} = h2_req(PushMock, get, <<"/activity">>),
+        #{<<"logs">> := Logs} = jiffy:decode(Body, [return_maps]),
+                                    
+        DeviceLogs = lists:filter(
+            fun(#{<<"device_token">> := Token}) ->
+                DeviceToken =:= Token
+            end, Logs),
+
+        case length(DeviceLogs) > 0 of 
+            true -> 
+                DeviceLogs;
+            false ->
+                throw({no_push_messages, DeviceToken})
+        end             
+     end).
+
+%% ----------------------------------
+%% Other helpers
+%% ----------------------------------
+
+create_room(Room, [Owner | Members], Config) ->
+    Domain = ct:get_config({hosts, mim, domain}),
+    create_room(Room, <<"muclight.", Domain/binary>>, Owner, Members,
+                                Config, <<"v1">>).
+
+bare_jid(JIDOrClient) ->
+    ShortJID = escalus_client:short_jid(JIDOrClient),
+    list_to_binary(string:to_lower(binary_to_list(ShortJID))).
+
+room_name(Config) ->
+    CaseName = proplists:get_value(case_name, Config),
+    <<"room_", (atom_to_binary(CaseName, utf8))/binary>>.
+
+is_offline(LUser, LServer) ->
+    case catch lists:max(rpc(ejabberd_sm, get_user_present_pids, [LUser, LServer])) of
+        {Priority, _} when is_integer(Priority), Priority >= 0 ->
+            false;
+        _ ->
+            true
+    end.
+
+gen_token() ->
+    integer_to_binary(binary:decode_unsigned(crypto:rand_bytes(16)), 24).
+
+become_unavailable(Client) ->
+    escalus:send(Client, escalus_stanza:presence(<<"unavailable">>)),
+    true = wait_for(timer:seconds(20), fun() ->
+        is_offline(lower(escalus_client:username(Client)), lower(escalus_client:server(Client)))
+    end). %% There is no ACK for unavailable status
+
+wait_for(TimeLeft, Fun) when TimeLeft < 0 ->
+    Fun();
+wait_for(TimeLeft, Fun) ->
+    Step = 500,
+    try
+        case Fun() of
+            ok -> ok;
+            true -> true;
+            {ok, _} = R -> R
+        end
+    catch
+        _:_ ->
+            timer:sleep(Step),
+            wait_for(TimeLeft - Step, Fun)
+    end.
+
+lower(Bin) when is_binary(Bin) ->
```

Comment:
Not really, I used it for two kinda generic binaries - username and server domain. As far I can see, the `escalus_utils:jid_to_lower/1` doesn't really care about that since it's implemented to handle any binary anyways. I'm not sure though that I should use function named `jid_to_lower` on a username for example.